### PR TITLE
Added support for tgw peering attachment

### PR DESCRIPTION
**Pull request by monty16597**
## Fixed
- Renamed class `VPCPeeringConnectionStatus` to `PeeringConnectionStatus ` as it can be used by other peering components.

## Added
- Support for `TransitGatewayPeeringAttachment `.
- API `CreateTransitGatewayPeeringAttachment`.
- API `DescribeTransitGatewayPeeringAttachments`.
- API `AcceptTransitGatewayPeeringAttachment`.
- API `DeleteTransitGatewayPeeringAttachment`.
- API `RejectTransitGatewayPeeringAttachment`.

## TerraformTestsReport
```
TestAccAWSEc2TransitGatewayPeeringAttachment PASS
TestAccAWSEc2TransitGatewayPeeringAttachmentAccepter FAIL
TestAccAWSEc2TransitGatewayPeeringAttachmentDataSource PASS
```

## Tested
- Terraform
```terraform

resource "aws_ec2_transit_gateway" "local" {
  tags = {
    Name = "Local TGW"
  }
}

resource "aws_ec2_transit_gateway" "peer" {
  tags = {
    Name = "Peer TGW"
  }
}

resource "aws_ec2_transit_gateway_peering_attachment" "example" {
  peer_account_id         = aws_ec2_transit_gateway.peer.owner_id
  peer_region             = "us-east-1"
  peer_transit_gateway_id = aws_ec2_transit_gateway.peer.id
  transit_gateway_id      = aws_ec2_transit_gateway.local.id

  tags = {
    Name = "TGW Peering Requestor"
  }
}

```

### CODE REVIEW

**Review by whummer — COMMENTED** [review]
LGTM, thanks @monty16597 👍 - added a few minor comments. Thanks

File: moto/ec2/utils.py [hunk 1]
```diff
@@ -675,3 +675,22 @@ def filter_iam_instance_profiles(iam_instance_profile_arn, iam_instance_profile_
...
+def describe_tag_filter(filters, instances):
+    instances = list(instances)
+    for key in filters:
+        if key.startswith('tag:'):
+            match = re.match(r'tag:(.*)', key)
+            if match:
+                tag_key_name = match.group(1)
+            for instance in instances:
+                need_delete = True
+                for tag in instance.get_tags():
```
Comment (whummer) [inline moto/ec2/utils.py:689]: Not sure I fully understand the logic in lines 688-695. Assuming `get_tags()` returns a list of two tags, say, `[tag1, tag2]`, and if the value of `tag2` matches, whereas the value of `tag1` does **not** match, then the `instance` would NOT get deleted. Is this the intended behavior here? (I would have expected that the instance should not be part of the result list if one of the tags is not matching)

File: moto/ec2/models.py [hunk 2]
```diff
@@ -6218,6 +6218,43 @@ def __init__(
...
+        )
+
+        self.accepter_tgw_info = {
+            "ownerId": peer_account_id,
+            "region": peer_region,
+            "transitGatewayId": peer_transit_gateway_id
+        }
+        self.requester_tgw_info = {
+            "ownerId": self.owner_id,
+            "region": "us-east-1",
```
[the remark this answers is not in the document]
Comment (whummer) [reply moto/ec2/models.py]: Please ignore - this comment was based on an earlier version of the code.

File: moto/ec2/models.py [hunk 3]
```diff
@@ -6301,6 +6338,68 @@ def describe_transit_gateway_vpc_attachments(self, transit_gateways_attachment_i
...
+        )
+
+        if transit_gateways_attachment_ids:
+            transit_gateways_attachments = [
+                transit_gateways_attachment
+                for transit_gateways_attachment in transit_gateways_attachments
+                if transit_gateways_attachment.id in transit_gateways_attachment_ids
+            ]
+
+        result = []
```
[the remark this answers is not in the document]
Comment (whummer) [reply moto/ec2/models.py]: Please ignore - this comment was based on an earlier version of the code.

File: moto/ec2/utils.py [hunk 4]
```diff
@@ -679,3 +679,23 @@ def filter_iam_instance_profiles(iam_instance_profile_arn, iam_instance_profile_
...
+def describe_tag_filter(filters, instances):
+    instances = list(instances)
+    for key in filters:
+        if not key.startswith('tag:'):
+            continue
+        match = re.match(r'tag:(.*)', key)
+        if match:
+            tag_key_name = match.group(1)
+        for instance in instances:
+            need_delete = True
```
Comment (monty16597) [inline moto/ec2/utils.py:693]: Let's say we have numbers of resource tags and we added tag filters, we want to find resources based on the tags which are supposed to be matched. We need to skip those resources whose resource tag key does not match with the filter tag but not the value.

So this line will delete any transit gateway by default. If none key match filter tags then it won't be present in the result.

File: moto/ec2/utils.py [hunk 5]
```diff
@@ -679,3 +679,23 @@ def filter_iam_instance_profiles(iam_instance_profile_arn, iam_instance_profile_
...
+            for tag in instance.get_tags():
+                if tag.get("key") == tag_key_name and tag.get('value') in filters.get(key):
```
Comment (monty16597) [inline moto/ec2/utils.py:695]: this line will set the flag for those resources who are matched with the filter tag.

File: moto/ec2/utils.py [hunk 6]
```diff
@@ -679,3 +679,23 @@ def filter_iam_instance_profiles(iam_instance_profile_arn, iam_instance_profile_
...
+                    need_delete = False
+                elif tag.get("key") == tag_key_name and tag.get('value') not in filters.get(key):
```
Comment (monty16597) [inline moto/ec2/utils.py:697]: if any other tag filter key matched with the resource tag key and value doesn't match then we need to remove those instances form the list.

**Pull request closed without merging**